### ISSUE TITLE
[hlsl-out] bad member names for generated input structs by naga 0.6

### ISSUE THREAD

**Issue by willemv** [gfx-rs/naga#1316]
When converting the imgui vertex shader from SPIR-V format to hlsl, the generated hlsl shader contains this input struct:

```hlsl
struct VertexInput_vs_main {
    float2 member : LOC0;
    float2 member : LOC1;
    float4 member : LOC2;
};
```

Note that the three member names have the same name `member`.

This is the vertex shader glsl source:

```glsl
#version 450

layout(set = 0, binding = 0) uniform View {
  mat4 u_Matrix;
};

layout(location = 0) in vec2 a_Pos;
layout(location = 1) in vec2 a_UV;
layout(location = 2) in vec4 a_Color;

layout(location = 0) out vec2 v_UV;
layout(location = 1) out vec4 v_Color;

// Built-in:
// vec4 gl_Position

void main() {
  v_UV = a_UV;
  v_Color = a_Color;
  gl_Position = u_Matrix * vec4(a_Pos.xy, 0.0, 1.0);
}
```

which is converted by shaderc (I think) to this SPIR-V file: [imgui.vert.spv.zip](https://github.com/gfx-rs/naga/files/7093921/imgui.vert.spv.zip)

**Comment by willemv** [comment]
I also tried updating to the latest code on the `v0.6` branch, which results in a similar bug but in the output struct. In f6a63b65daf3a36d404ffbcd617eeb8f8576847b I think the vertex shader input was changed to not use structs, but the output was converted to a struct.

In any case, the latest code on the `v0.6` branch generates output structs with identical member names.

**Comment by kvark** [comment]
Thanks for the report!
This was fixed in #1303, which is awaiting to be released in a patch.
